core/src/test/scala/org/apache/spark/scheduler/TaskResultGetterSuite.scala

```diff
@@ -81,6 +81,16 @@ class TaskResultGetterSuite extends SparkFunSuite with BeforeAndAfter with Local
   // as we can make it) so the tests don't take too long.
   def conf: SparkConf = new SparkConf().set("spark.akka.frameSize", "1")
 
+  test("Kryo serializer for TaskResult") {
+    val conf1 = new SparkConf(false)
+    conf1.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+    conf1.set("spark.kryoserializer.buffer", "1m")
+    conf1.set("spark.kryoserializer.buffer.max", "2m")
+    sc = new SparkContext("local", "test", conf1)
+    val result = sc.parallelize(Seq(1), 1).map(x => 2 * x).reduce((x, y) => x)
+    assert(result === 2)
```

Comment:
All I'm suggesting is a test that looks like the one added here:
https://github.com/apache/spark/commit/de0278286cf6db8df53b0b68918ea114f2c77f1f

In general tests should be as fine-grained as possible. Right now this just does an end-to-end test, which may be sufficient but is quite brittle. E.g. imagine if we change the name of the config or the serializer, then this won't actually test what we want.